### ISSUE TITLE
Additional Performance improvements on "Making our Network More Efficient"

### ISSUE THREAD

**Issue by kirilcvetkov92** [udacity/deep-learning-v2-pytorch#128]
As a part of # Sentiment Classification & How To "Frame Problems" for a Neural Network,
**Project 5: Making our Network More Efficient**,
I wonder why the author use the slow python for loops for sum which can be replaced with numpy operations that are wrapped in c++.

For example :
for index in review:
     self.layer_1 +=self.weights_0_1[index]

can be replaced with :
self.layer_1+=self.weights_0_1[review].sum(axis=0)

*On my PC I'm achieving 1.26x Speedup*, :slightly_smiling_face: since this is "optimization" part, I see sensible reason for eliminating the slow python for loops.

Note (Similar scenario appears during the weights update)
for index in review:
               self.weights_0_1[index] -= layer_1_delta[0] *  self.learning_rate
can be replaced with
self.weights_0_1[review] -= layer_1_delta[0] * self.learning_rate

*Achieving total 2.89X speedup 5532 (reviews/sec)*

**Comment by abhiojha8** [comment]
Thanks for the feedback. I agree that numpy would be faster. Happy to review and approve a PR for this.
